.github/workflows/relay-test.yml

```diff
@@ -46,37 +47,57 @@ jobs:
           username: ${{ secrets.DOCKER_HUB_USERNAME }}
           password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
 
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
       - name: Set up Docker Buildx
         id: buildx
         uses: docker/setup-buildx-action@v2
 
-      - name: Install protoc
-        run: |
-          PROTOC_ZIP=protoc-3.14.0-linux-x86_64
-          curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v3.14.0/$PROTOC_ZIP.zip
-          unzip -o $PROTOC_ZIP.zip -d $HOME/.local
-          chmod +x $HOME/.local/bin/protoc
-          rm -f $PROTOC_ZIP.zip
-
-      - name: Build relay
-        working-directory: tools/relay
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+        with:
+          version: '3.x'
+
+      - name: Setup go dependencies
         run: |
-          set -x
           go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.26
           go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.1
-          chmod +x $(go env GOPATH)/bin/*
-          export PATH="$PATH:$(go env GOPATH)/bin"
+          # chmod +x $(go env GOPATH)/bin/*
+          # export PATH="$PATH:$(go env GOPATH)/bin"
+
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          push: true
+          file: tools/relay/Dockerfile
+          platforms: ${{ env.PLATFORMS }}
+          target: relay
+          tags: vhiveease/relay:latest
+          context: .
+
+
 
-          make all
+  integration-tests:
+    name: Relay Integration Tests
+    needs: build-image
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v3
 
-      - name: docker-compose up
+      - name: Start container
         working-directory: benchmarks/aes
         run: |
           set -x
 
           docker-compose -f ./yamls/docker-compose/dc-aes-go.yaml up &> log_file &
           sleep 15
 
+      - name: Install Protoc
+        uses: arduino/setup-protoc@v1
+        with:
+          version: '3.x'
```

Comment:
I'd recomment to fix the version more carefully, e.g., 3.9 3.16 (please check the logs), to avoid unexpected test breakages

Reply:
Agree! Thanks!